gms/gossiper.cc

```diff
@@ -2172,32 +2173,31 @@ future<> gossiper::add_local_application_state(std::initializer_list<std::pair<a
 //
 future<> gossiper::add_local_application_state(std::list<std::pair<application_state, versioned_value>> states) {
     if (states.empty()) {
-        return make_ready_future<>();
+        co_return;
     }
-    return container().invoke_on(0, [states = std::move(states)] (gossiper& gossiper) mutable {
-        return seastar::async([g = gossiper.shared_from_this(), states = std::move(states)]() mutable {
-            auto& gossiper = *g;
+    try {
+        co_await container().invoke_on(0, [&] (gossiper& gossiper) mutable -> future<> {
```

Comment:
coroutine::lambda()?

Reply:
I understand we need coroutine::lambda for continuations.
Is it needed for invoke_on / submit_to calls as well?

Is it needed when the lambda has no captures?

Reply:
It's capturing &, so one variable.

I looked at submit_to() and it moves the lambda to a safe place, so it's not in the same category as a continuation. We should mark it as safe for continuations.